topologywatcher_http.go

```diff
@@ -0,0 +1,87 @@
+package gocbcorex
+
+import (
+	"context"
+	"net/http"
+
+	"go.uber.org/zap"
+)
+
+type TopologyWatcherHttpConfig struct {
+	HttpRoundTripper http.RoundTripper
+	Endpoints        []string
+	UserAgent        string
+	Authenticator    Authenticator
+	BucketName       string
+}
+
+type TopologyWatcherHttpOptions struct {
+	Logger *zap.Logger
+}
+
+type TopologyWatcherHttp struct {
+	logger *zap.Logger
+
+	configWatcher *ConfigWatcherHttp
+	watcherCancel func()
+}
+
+func NewTopologyWatcherHttp(config *TopologyWatcherHttpConfig, opts *TopologyWatcherHttpOptions) (*TopologyWatcherHttp, error) {
```

Comment:
I'd like it more if this used DI and accepted a config watcher, but reconfigure would then have to do the same - at which point the caller has to own a config watcher and reconfigure that itself.

Reply:
What is the difference between a configWatch and a topologyWatcher?  Why can't they be one in the same?

Reply:
They could, I just kind of liked the separation of deduplication, allowing a caller to manager it themselves if desired.